checks/allprocesses_windows.go

```diff
@@ -0,0 +1,306 @@
+		pid := pe32.Th32ProcessID
+		ppid := pe32.Th32ParentProcessID
+
+		// 0x1000 is PROCESS_QUERY_LIMITED_INFORMATION, but that constant isn't
+		// defined in syscall
+		procHandle, err := syscall.OpenProcess(0x1000, false, uint32(pid))
+		if err != nil {
+			continue
+		}
+		defer syscall.CloseHandle(procHandle)
+
+		var CPU syscall.Rusage
+		if err = syscall.GetProcessTimes(procHandle, &CPU.CreationTime, &CPU.ExitTime, &CPU.KernelTime, &CPU.UserTime); err != nil {
+			continue
+		}
+
+		var handleCount uint32
+		if err = getProcessHandleCount(procHandle, &handleCount); err != nil {
+			continue
+		}
+
+		var pmemcounter process.PROCESS_MEMORY_COUNTERS
+		if err = getProcessMemoryInfo(procHandle, &pmemcounter); err != nil {
+			continue
+		}
+
+		// shell out to getprocessiocounters for io stats
+		var ioCounters IO_COUNTERS
+		if err = getProcessIoCounters(procHandle, &ioCounters); err != nil {
+			continue
+		}
+		ctime := CPU.CreationTime.Nanoseconds() / 1000000
+
+		exebase := convert_windows_string(pe32.SzExeFile[:])
+		cmdbase := *fromWmi[pid].CommandLine
+		var parsedargs []string
+		if len(cmdbase) == 0 {
+			parsedargs = append(parsedargs, exebase)
+		} else {
+			parsedargs = parseCmdLineArgs(cmdbase)
+		}
+
+		utime := float64((int64(CPU.UserTime.HighDateTime) << 32) | int64(CPU.UserTime.LowDateTime))
+		stime := float64((int64(CPU.KernelTime.HighDateTime) << 32) | int64(CPU.KernelTime.LowDateTime))
+		username, err := get_username_for_process(procHandle)
+		procs[int32(pid)] = &process.FilledProcess{
+			Pid:     int32(pid),
+			Ppid:    int32(ppid),
+			Cmdline: parsedargs,
+			CpuTime: cpu.TimesStat{
+				User:      utime,
+				System:    stime,
+				Timestamp: time.Now().UnixNano(),
+			},
+
+			CreateTime:  ctime,
+			OpenFdCount: int32(handleCount),
+			//Name
+			// Status
+			// UIDS
+			// GIDs
+			NumThreads:  int32(pe32.CntThreads),
+			CtxSwitches: &process.NumCtxSwitchesStat{},
+			MemInfo: &process.MemoryInfoStat{
+				RSS:  pmemcounter.WorkingSetSize,
+				VMS:  pmemcounter.QuotaPagedPoolUsage,
+				Swap: 0, // it's unclear there's a Windows measurement of swap file usage
+			},
+			//Cwd
+			Exe: exebase,
+			IOStat: &process.IOCountersStat{
+				ReadCount:  ioCounters.ReadOperationCount,
+				WriteCount: ioCounters.WriteOperationCount,
+				ReadBytes:  ioCounters.ReadTransferCount,
+				WriteBytes: ioCounters.WriteTransferCount,
+			},
+			Username: username,
+		}
+	}
+	return procs, nil
+}
+
+func get_username_for_process(h syscall.Handle) (name string, err error) {
+	name = ""
+	err = nil
+	var t syscall.Token
+	err = syscall.OpenProcessToken(h, syscall.TOKEN_QUERY, &t)
+	if err != nil {
+		log.Debugf("Failed to open process token %v", err)
+		return
+	}
+	defer t.Close()
+	tokenUser, err := t.GetTokenUser()
+
+	user, domain, _, err := tokenUser.User.Sid.LookupAccount("")
+	return domain + "\\" + user, err
+
+}
+
+func convert_windows_string(winput []uint16) string {
```

Comment:
convertWindowsString (I see this function was copied - but might as well fix it)